File: golang/tests/pkg/agentbay/adb_test.go
package agentbay_test

import (
	"fmt"
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
)

func TestAdb_Shell(t *testing.T) {
	// Initialize AgentBay client
	apiKey := getTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	// Create a session
	fmt.Println("Creating a new session for ADB testing...")
	session, err := agentBay.Create(nil)
	if err != nil {
		t.Fatalf("Error creating session: %v", err)
	}
	t.Logf("Session created with ID: %s", session.SessionID)

	// IMPORTANT: Ensure cleanup of the session after test
	defer func() {
		fmt.Println("Cleaning up: Deleting the session...")
		err := agentBay.Delete(session)
		if err != nil {
			t.Logf("Warning: Error deleting session: %v", err)
		} else {
			t.Log("Session successfully deleted")
		}
	}()

	// Test Adb shell command execution
	if session.Adb != nil {
		fmt.Println("Executing ADB shell command...")
		response, err := session.Adb.Shell("ls /sdcard")
		if err != nil {
			t.Logf("Note: ADB shell execution failed: %v", err)
		} else {
			t.Logf("ADB shell execution result: %s", response)
			// Check if response contains "tool not found"
			if containsToolNotFound(response) {
				t.Errorf("Adb.Shell returned 'tool not found'")
			}
		}

		// Add more ADB-related tests
		fmt.Println("Executing ADB shell command to check device properties...")
		propResponse, err := session.Adb.Shell("getprop")
		if err != nil {
			t.Logf("Note: ADB getprop execution failed: %v", err)
		} else {
			t.Logf("ADB getprop execution result length: %d bytes", len(propResponse))
			// Check if response contains "tool not found"
			if containsToolNotFound(propResponse) {
				t.Errorf("Adb.Shell returned 'tool not found'")
			}
		}
	} else {
		t.Logf("Note: Adb interface is nil, skipping ADB test")
	}
}

File: typescript/tests/unit/adb.test.ts
import { AgentBay, Session } from '../../src';
import { getTestApiKey, containsToolNotFound } from '../utils/test-helpers';

// Define test runner types if they're not available
declare function describe(name: string, fn: () => void): void;
declare function beforeEach(fn: () => void | Promise<void>): void;
declare function afterEach(fn: () => void | Promise<void>): void;
declare function it(name: string, fn: () => void | Promise<void>): void;
declare function expect(actual: any): any;

describe('Adb', () => {
  let agentBay: AgentBay;
  let session: Session;
  
  beforeEach(async () => {
    const apiKey = getTestApiKey();
    agentBay = new AgentBay({ apiKey });
    session = await agentBay.create();
  });
  
  afterEach(async () => {
    // Clean up the session
    try {
      await agentBay.delete(session);
      console.log(`Session deleted successfully: ${session.sessionId}`);
      
    } catch (error) {
      console.log(`Warning: Error deleting session: ${error}`);
    }
  });
  
  describe('shell', () => {
    it('should execute ADB shell commands', async () => {
      if (session.adb) {
        console.log('Executing ADB shell command...');
        try {
          const response = await session.adb.shell('ls /sdcard');
          console.log(`ADB shell execution result: ${response}`);
          
          // Check if response contains "tool not found"
          expect(containsToolNotFound(response)).toBe(false);
          
          // We don't check the specific content as it depends on the device
          expect(response).toBeDefined();
        } catch (error) {
          console.log(`Note: ADB shell execution failed: ${error}`);
          // Don't fail the test if ADB is not supported
        }
        
        // Test another ADB command
        try {
          console.log('Executing ADB shell command to check device properties...');
          const propResponse = await session.adb.shell('getprop');
          console.log(`ADB getprop execution result length: ${propResponse.length} bytes`);
          
          // Check if response contains "tool not found"
          expect(containsToolNotFound(propResponse)).toBe(false);
          
          // We don't check the specific content as it depends on the device
          expect(propResponse).toBeDefined();
          expect(propResponse.length).toBeGreaterThan(0);
        } catch (error) {
          console.log(`Note: ADB getprop execution failed: ${error}`);
          // Don't fail the test if ADB is not supported
        }
      } else {
        console.log('Note: Adb interface is nil, skipping ADB test');
      }
    });
  });
});
